Design.md

```diff
@@ -32,71 +32,73 @@ Here are the IELE design decisions described from the perspective of differences
 
-* The CREATE instruction is modified. Instead of taking a memory start and a memory width containing initialization code, the CREATE instruction takes an immediate data segment offset, and an arbitrary number of register arguments to the constructor. If the data segment offset is not the offset of a CONTRACT instruction, the contract is malformed.
-* The CONTRACT instruction contains a sequence of instructions in binary form as its string value. If the contract specified is malformed, the contract containing the instruction is malformed.
-* The CREATE function copies the contract contained in its respective CONTRACT instruction into the code storage of the account that contains the created contract. Then the function at function label zero is invoked in the context of the created account to initialize the contract. It is passed the values in the registers as arguments in the same fashion as a CALL, CALLCODE, DELEGATECALL, or STATICCALL instruction. If this function returns a value, the contract is malformed.
-* A separate instruction, COPYCREATE, is created for the case of copying an entire contract, or duplicating your own contract. The COPYCREATE instruction behaves like CREATE, except that instead of taking an immediate data segment offset, it takes a register operand containing the address, and copies the entire code storage of that contract.
-* Both instructions, like before, return the address of the created contract on success and zero on failure.
-* Because we no longer actually need to address portions of code by program counter in order to create contracts, we remove the CODECOPY and EXTCODECOPY instructions. Because these are also the only remaining instructions which require the code to be addressed by a particular byte offset, we also remove the PC instruction.
-* Transactions that create a contract now explicitly provide the binary data of the contract instead of the binary data of the initialization function. Contracts created in this way are not subject to the restriction that dynamic contract data be inserted, and still invoke the function at function label 0 to initialize the contract. If the contract is malformed, the transaction fails.
+## Contract Creation
 
-## Code Size
+Unlike EVM, both for ease of verification and due to security concerns, we disallow creation of dynamic code for dynamically created contracts, and instead restrict dynamically-created-contracts to code that consists of some previously validated contract.
 
-The maximum length of a IELE contract is expanded to 65536 bytes.
+* The `create` instruction is modified from its EVM counterpart. Instead of taking a memory start and a memory width containing initialization code, the `create` instruction takes a contract name, and an arbitrary number of register arguments to be passed to the `@init` function of the contract to be created. If the contract name is not an externally declared contract name, the contract is malformed.
```

Comment:
This raises the issue of whether we want to have some static checker for malformedness.  I would put that at the end in the to-do list.

Reply:
Yes, I think @dwightguth has mentioned the same, that he wants to have one checker for well-formed IELE programs in the future. Currently the semantics assumes only well-formed programs.